### PR TITLE
Configure VPC endpoint services for container clusters

### PR DESCRIPTION
**Pull request by jonmv**
@freva and perhaps @mpolden please review.

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/lb/SharedLoadBalancerService.java [hunk 1]
```diff
@@ -28,12 +29,14 @@ public SharedLoadBalancerService(String vipHostname) {
 
     @Override
     public LoadBalancerInstance create(LoadBalancerSpec spec, boolean force) {
+        if (spec.settings() != LoadBalancerSettings.empty) throw new IllegalArgumentException("custom load balancer settings are not supported with " + getClass());
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/lb/SharedLoadBalancerService.java:32]: Should be `equals`, so non-empty settings always throws.

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/ServiceRegistry.java [hunk 2]
```diff
@@ -57,6 +58,8 @@ public interface ServiceRegistry {
 
     NameService nameService();
 
+    VpcEndpointService vpcEndpointService();
```
Comment (mpolden) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/ServiceRegistry.java:61]: Consider a more generic name. `PrivateEndpointService`?

**Review by mpolden — COMMENTED** [review]
PR title looks wrong. 🙂

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/lb/SharedLoadBalancerService.java (hunk 1 above)
Comment (jonmv) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/lb/SharedLoadBalancerService.java:32]: Yes, replaced with `isEmpty()` method.

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/ServiceRegistry.java (hunk 2 above)
Comment (jonmv) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/ServiceRegistry.java:61]: Isn't VPC "virtual private cloud", and that concept is used in other clouds too? 
It should in any case be a `...EndpointServiceService` >_<

**Review by freva — APPROVED** [review]

**Pull request merged**